commons/src/test/java/com/powsybl/commons/xml/XmlUtilTest.java

```diff
@@ -43,4 +45,29 @@ public void readUntilEndElementWithDepthTest() throws XMLStreamException {
         }
         assertEquals(ImmutableMap.of("a", 0, "b", 1, "c", 2, "d", 1), depths);
     }
+
+    @Test
+    public void nestedReadUntilEndElementWithDepthTest() throws XMLStreamException {
+        Map<String, Integer> depths = new HashMap<>();
+        try (StringReader reader = new StringReader(XML)) {
+            XMLStreamReader xmlReader = XMLInputFactory.newInstance().createXMLStreamReader(reader);
+            try {
+                XmlUtil.readUntilEndElementWithDepth("a", xmlReader, elementDepth -> {
+                    depths.put(xmlReader.getLocalName(), elementDepth);
+                    // consume b and c
+                    if (xmlReader.getLocalName().equals("b")) {
+                        try {
+                            XmlUtil.readUntilEndElement("b", xmlReader, () -> {});
```

Comment:
Coding style: checkstyle failed if there is no space after an opening brace. You can write something like this:
```
XmlUtil.readUntilEndElement("b", xmlReader, () -> { });
```

Reply:
fixed